gemd/demo/cake.py

```diff
@@ -113,6 +115,11 @@ def make_cake_templates():
         description="The chemical formula of a material",
         bounds=CompositionBounds(components=EmpiricalFormula.all_elements())
     )
+    tmpl["Molecular Structure"] = PropertyTemplate(
+        name="Molecular Structure",
+        description="The molecular structure of the material",
+        bounds=MolecularStructureBounds()
+    )
```

Comment:
How is this template used?

Reply:
It appears (or should be appearing) as a PropertyAndCondition on the MaterialSpec for salt and for sugar.  The former is represented as an InChI, the latter as a SMILES.

Reply:
How are the `PropertyAndCondition` objects linked back to the template? I see, for example,
`Property(name='SMILES', value=Smiles("C(C1C(C(C(C(O1)OC2(C(C(C(O2)CO)O)O)CO)O)O)O)O")))`, but shouldn'T that property be linked to the `PropertyTemplate`?

Reply:
Yes.  Yes it should.